### PR TITLE
Refactor/improve namespace hierarchy

### PR DESCRIPTION
**Pull request by cdoehn**
## Planned namespaces

* [x] api
* [x] configuration
* [x] protocol_layer
* [x] -----scanner_events
* [x] data_conversion_layer
* [x] -----monitoring frame
* [x] ----------diagnostic
* [x] -----scanner reply
* [x] -----start_request
* [x] -----raw_processing
* [x] communication layer

## namespaces to be removed

* [x] additional_field
* [x] format error
* [x] constants

## add subdirectories in file structure
* [x] api
* [x] configuration
* [x] protocol_layer
* [x] data_conversion_layer
* [x] communication layer
* [x] util

## Description

These changes Fix # | allow the user to | improve | ...

### Things to add, update or check by the maintainers before this PR can be merged.

* [x] Public api function documentation
* [x] Architecture documentation reflects actual code structure
* [x] ~~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~~
* [x] ~~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~~
* [x] ~~Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))~~
* [x] ~~Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))~~
* [x] ~~CHANGELOG.rst updated~~
* [x] ~~Copyright headers~~
* [x] ~~Examples~~

### Review Checklist
* [x] Soft- and hardware architecture (diagrams and description)
* [x] ~~Test review (test plan and individual test cases)~~
* [x] Documentation describes purpose of file(s) and responsibilities
* [x] ~~Code (coding rules, style guide)~~

~~### Release planning (please answer)~~
* [x] ~~When is the new feature released?~~
* [x] ~~Which dependent packages do have to be released simultaneously?~~

### CODE REVIEW

**Comment by rfeistenauer** [comment]
Some thoughts:
- LaserScan could be put into the api namespace since this is the one the user gets as output correct?
- ScanRange could be put into configuration namespace (or together with the content of that namespace into api, not sure if configuration namespace is necessary)
- InternalScannerReplyError and monitoring_frame namespace are missing some brief documentation
- scanner_v2.h is in include folder and include/api folder?
- Maybe Move Barrier, Watchdoc and TenthOfDegree into an util namespace?
- separate folders could be positive for navigating in source files, if we keep them I would create an folder for all top lvl namespaces then (api, communication layer, ...)

File: CHANGELOG.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ Forthcoming
 
 0.1.2 (2020-11-24)
 ------------------
-* Add Scanner ports in ScannerConfiguration
+* Add Scanner ports in configuration::ScannerConfiguration
```
Comment (martiniil) [inline CHANGELOG.rst:32]: I think you usually do not change past entries in the changelog.

Comment (martiniil) [reply CHANGELOG.rst:32]: Generally, I like the approach.

Comment (martiniil) [reply CHANGELOG.rst:32]: Minor comments:
- `laserscan.cpp` and `.h` are currently duplicate
- I think the `ScanValidator` in `complete_scan_validator.h` belongs to the protocol_layer
- `format_range` belongs into util namespace/subfolder
- in general all headers should be sorted into the subfolders except maybe `core.h`. What do you think?

Comment (cdoehn) [reply CHANGELOG.rst]: I think ScanValidator better belongs to MonitoringFrame...

Comment (martiniil) [reply CHANGELOG.rst]: Are you sure? This is what the `ScanValidator` does:
```
//! @brief Validates complete scan rounds and detects if MonitoringFrames are missing for a complete scan or
//! if to many MonitoringFrames were received.
```

Comment (cdoehn) [reply CHANGELOG.rst]: ok

**Review by martiniil — APPROVED** [review]
I would omit the namespace `constants` in `configuration/default_parameters.h`. But it is not required.

I really like that this is more structured now. My feeling is that we have not overdone it. However it is just a feeling. I would stick with it for now.

**Pull request merged**